### PR TITLE
Issue 26 feat/app create feed screen

### PR DESCRIPTION
**Pull request by T0ffik**

### CODE REVIEW

**Comment by Mariusz-web** [comment]
Can we move the date bar or it will be static and show a given week?
Something deploy is not loading but I think the dates at 425px are two lines. You could try to reduce these somehow to always keep them on one line :)

**Comment by Mariusz-web** [comment]
Can we move the date bar or it will be static and show a given week?
Something deploy is not loading but I think the dates at 425px are two lines. You could try to reduce these somehow to always keep them on one line :)

**Review by Mariusz-web — APPROVED** [review]

File: apps/client/src/app/components/Dish/Dish.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
+import styled from '@emotion/styled';
+import { Add, ExpandLess, ExpandMore } from '@mui/icons-material';
+import { IconButton, Typography } from '@mui/material';
+import { Box } from '@mui/system';
+import { useState } from 'react';
+import { getDay } from '../../mocks/dishesOfTheDay';
+import { dish, product } from '../../utils/types/dish.types';
+import { Product } from '../Product';
+
+type props = {
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Dish/Dish.tsx:10]: Capital letter

File: apps/client/src/app/components/Dish/Dish.tsx [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    name: string;
+    dish: dish;
+};
+
+export const Dish = ({ dish, name }: props) => {
+    const [seeMore, setSeeMore] = useState(false);
+    const DishBox = styled(Box)`
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Dish/Dish.tsx:17]: no point of declaring that inside of the component

File: apps/client/src/app/components/Dish/Dish.tsx [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        flex-direction: column;
+        width: 100%;
+    `;
+    return (
+        <>
+            <DishBox>
+                <Box>
+                    <Typography variant='h6'>
+                        {name}{' '}
+                        {dish.kcal !== '0' ? (
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Dish/Dish.tsx:42]: kinda sketchy, that probably wouldn't be string

File: apps/client/src/app/components/Dish/Dish.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+                        {dish.kcal !== '0'
+                            ? `B: ${dish.B}g T: ${dish.T}g W: ${dish.W}g`
+                            : null}
+                    </Typography>
+                </Box>
+                <AddButton>
+                    <Add />
+                </AddButton>
+            </DishBox>
+            {seeMore ? (
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Dish/Dish.tsx:61]: `seeMore && dish.Products`

Also huge inline connditional render is kinda bad

File: apps/client/src/app/components/Footer/Footer.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+};
+
+type Props = {
+    eaten: Eaten;
+};
+
+export const Footer = ({ eaten }: Props) => {
+    const goal = dailyGoal();
+    return (
+        <div
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Footer/Footer.tsx:17]: div?

File: apps/client/src/app/components/Product/Product.tsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+import { Cancel, ConstructionOutlined } from '@mui/icons-material';
+import { product } from '../../utils/types/dish.types';
+
+type props = {
+    product: product;
+};
+
+export const Product = ({ product }: props) => {
+    const handleDetele = () => console.log('deleting product'); // TODO connect with API
+    const ProductBox = styled(Box)`
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Product/Product.tsx:13]: same, we don't need to declare that inside of the componetn

File: apps/client/src/app/components/Product/Product.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
+import styled from '@emotion/styled';
+import { Box } from '@mui/system';
+import { IconButton, Typography } from '@mui/material';
+import { Cancel, ConstructionOutlined } from '@mui/icons-material';
+import { product } from '../../utils/types/dish.types';
+
+type props = {
```
Comment (Veryyapeee) [inline apps/client/src/app/components/Product/Product.tsx:7]: capital letters

File: apps/client/src/app/sceens/Feed/Feed.tsx [hunk 8]
```diff
@@ -1,3 +1,108 @@
+import { PaginationItem, Typography, Pagination } from '@mui/material';
+import { convertDate } from '../../utils/convertDate';
+import styled from '@emotion/styled';
+import { useMemo, useState } from 'react';
+import { Dish, Footer } from '../../components';
+import { getDay } from '../../mocks/dishesOfTheDay';
+
 export const Feed = () => {
-    return <div>Feed</div>;
+    let dupa = -1;
```
Comment (Veryyapeee) [inline apps/client/src/app/sceens/Feed/Feed.tsx:9]: ;-;

File: apps/client/src/app/sceens/Feed/Feed.tsx [hunk 9]
```diff
@@ -1,3 +1,108 @@
...
+    const [dayOfWeek, setDayofWeek] = useState(new Date().getDay());
+    const date = useMemo(() => new Date(), [new Date()]);
```
Comment (Veryyapeee) [inline apps/client/src/app/sceens/Feed/Feed.tsx:11]: what's that? XD

1. You don't need to memoize new Date(),
2. Also why are you putting the results of new Date() into dependencies array? xd

File: apps/client/src/app/sceens/Feed/Feed.tsx [hunk 10]
```diff
@@ -1,3 +1,108 @@
...
+        text-align: center;
+    `;
+    const PaginationStyled = styled(Pagination)`
+        .MuiPagination-ul {
+            background-color: #4caf50;
+            display: flex;
+            justify-content: space-between;
+        }
+    `;
+    const dayOfTheWeek = (day: number) => {
```
Comment (Veryyapeee) [inline apps/client/src/app/sceens/Feed/Feed.tsx:56]: that could be a utility function outside of the component

File: apps/client/src/app/sceens/Feed/Feed.tsx [hunk 11]
```diff
@@ -1,3 +1,108 @@
...
+        fat:
+            parseInt(day.Breakfast.T) +
+            parseInt(day.Dinner.T) +
+            parseInt(day.Supper.T),
+        carbs:
+            parseInt(day.Breakfast.W) +
+            parseInt(day.Dinner.W) +
+            parseInt(day.Supper.W),
+    };
+    const PaginationDay = styled(PaginationItem)`
```
Comment (Veryyapeee) [inline apps/client/src/app/sceens/Feed/Feed.tsx:41]: same as before

File: apps/client/src/app/sceens/Feed/Feed.tsx [hunk 12]
```diff
@@ -1,3 +1,108 @@
...
+        }
+    };
+    return (
+        <>
+            <PaginationStyled
+                hideNextButton={true}
+                hidePrevButton={true}
+                count={7}
+                renderItem={(item) => {
+                    ++dupa;
```
Comment (Veryyapeee) [inline apps/client/src/app/sceens/Feed/Feed.tsx:83]: dupa dupa

**Review by Veryyapeee — APPROVED** [review]
Approved cuz of no time, generally quality of this code is not at the highest lvl, I guess we don't have time, just remove the "dupa's" xd

**Pull request closed without merging**